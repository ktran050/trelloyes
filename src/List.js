import React from "react";
import "./List.css";
import Card from "./Card.js";

class List extends React.Component {
  render() {
    const cardsList = this.props.cards.map((card) => (
      <Card title={card.title} content={card.content} />
    ));
    return (
      <section className="List">
        <header className="List-header">
          <h2>{this.props.header}</h2>
        </header>
        <div className="List-cards">{cardsList}</div>
      </section>
    );
  }
}

export default List;

/* The List component should render markup matching the design.html:
  a section with a class of 'List'
    containing a header
    and div with class 'List-cards'.
The List component accepts 2 props: header and cards.
header is a string of the header of the card to render.
cards is an array of card objects. Each object should have a title and content.
These props will be passed in for each List from the App component.
The List should render a Card component for each of the cards in the cards array prop.
Each instance of the Card component should be passed 2 props (and a key). The 2 props are title and content.
The title prop is a string for the Card's title.
The content prop is a string of the Card's content.

*/
