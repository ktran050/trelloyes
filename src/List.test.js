import React from "react";
import ReactDOM from "react-dom";
import List from "./List.js";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");
  const arr = {
    a: { id: "a", title: "First card", content: "lorem ipsum" },
    b: { id: "b", title: "second card", content: "lorem ipsum" },
  };
  ReactDOM.render(
    <List key="someKey" header="smoke" cards={[arr["a"], arr["b"]]} />,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders in the manner expected", () => {
  const tree = renderer
    .create(
      <List
        header="smoke"
        cards={[
          { id: "a", title: "First card", content: "lorem ipsum" },
          { id: "b", title: "second card", content: "lorem ipsum" },
        ]}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
