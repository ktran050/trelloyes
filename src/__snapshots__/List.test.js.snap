// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders in the manner expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      smoke
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        second card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
</section>
`;
